hollow/src/main/java/com/netflix/hollow/api/producer/validation/ObjectModificationValidator.java

```diff
@@ -0,0 +1,122 @@
+		this.typeName = typeName;
+		this.filter = filter;
+        this.statusBuilder = SingleValidationStatus.builder(ObjectModificationValidator.class.getSimpleName())
+                .addAdditionalInfo("typeName", typeName);
+    }
+
+    @Override
+    @SuppressWarnings("unchecked")
+	public void validate(ReadState readState) throws ValidationException {
+        try {
+            HollowTypeReadState typeState = readState.getStateEngine().getTypeState(typeName);
+            if (typeState == null) {
+                throw new ValidationException("Cannot execute ObjectModificationValidator on missing type "
+                        + typeName);
+            } else if (typeState.getSchema().getSchemaType() != SchemaType.OBJECT) {
+                throw new ValidationException("Cannot execute ObjectModificationValidator on type " + typeName
+                        + " because it is not a HollowObjectSchema - it is a "
+                        + typeState.getSchema().getSchemaType());
+            }
+            BitSet latestOrdinals = typeState.getPopulatedOrdinals();
+            BitSet previousOrdinals = typeState.getPreviousOrdinals();
+            if (previousOrdinals.isEmpty()) {
+                statusBuilder.withMessage("Skipping " + ObjectModificationValidator.class.getSimpleName()
+                        + " because previous cycle has no " + typeName).withStatus(Status.SKIP);
+                return;
+            }
+            BitSet removedAndModified = calculateRemovedAndModifiedOrdinals(latestOrdinals, previousOrdinals);
+            if (removedAndModified.isEmpty()) {
+                statusBuilder.withMessage("Skipping " + ObjectModificationValidator.class.getSimpleName()
+                        + " because " + typeName + " has no removals or modifications").withStatus(Status.SKIP);
+                return;
+            }
+            HollowObjectTypeReadState objectTypeState = (HollowObjectTypeReadState) typeState;
+            PrimaryKey key = objectTypeState.getSchema().getPrimaryKey();
+            // this is guaranteed to give us items from the most recent cycle, not the last one
+            HollowPrimaryKeyIndex index = new HollowPrimaryKeyIndex(readState.getStateEngine(), key);
+            int fromOrdinal = removedAndModified.nextSetBit(0);
+            while (fromOrdinal != HollowConstants.ORDINAL_NONE) {
+                Object[] candidateKey = index.getRecordKey(fromOrdinal);
+                int matchedOrdinal = index.getMatchingOrdinal(candidateKey);
+                if (matchedOrdinal != HollowConstants.ORDINAL_NONE) {
+                    T fromObject = (T) new GenericHollowObject(objectTypeState, fromOrdinal);
+                    T toObject = (T) new GenericHollowObject(objectTypeState, matchedOrdinal);
+                    if (!filter.apply(fromObject, toObject)) {
+                        throw new ValidationException("Validation failed for key " + Arrays.toString(candidateKey));
```

Comment:
I wonder if validation should fail fast or should test every modified object and accumulate the failures?